blockchain/chain_sync/src/chain_muxer.rs

```diff
@@ -271,19 +268,9 @@ where
 
             // Update the peer metadata based on the response
             match response {
-                Some(Ok(_res)) => {
+                Some(_) => {
                     network.peer_manager().log_success(peer_id, dur).await;
                 }
-                Some(Err(why)) => match why {
-                    RequestResponseError::ConnectionClosed
-                    | RequestResponseError::DialFailure
-                    | RequestResponseError::UnsupportedProtocols => {
-                        network.peer_manager().mark_peer_bad(peer_id).await;
```

Comment:
This is not an issue removing this code? Or is `mark_peer_bad` done elsewhere?

Reply:
This logic is never hit, in the old code it always sends Ok(response) to the channel: https://github.com/ChainSafe/forest/pull/2498/files#diff-9bebebe2f8a520466a9fe8ca9c212cc919b05b9ca67c44773742cf31f96b1a29L629
```rust
if tx.send(Ok(response)).is_err() {
    warn!("Fail to send Hello response");
}
```

Reply:
You are right, I should have added this logic to here, now fixed: https://github.com/ChainSafe/forest/pull/2498/files#diff-9bebebe2f8a520466a9fe8ca9c212cc919b05b9ca67c44773742cf31f96b1a29R627